### ISSUE TITLE
On-ride photo profit calculation assumes that every guest buys one

### ISSUE THREAD

**Issue by Gymnasiast** [OpenRCT2/OpenRCT2#10946]
**OS:** Ubuntu 20.04
**Version:** 0.2.4+
**Commit/Build:** fd2e3f7b08a810d5c20466d794326e37c46f4b1e

Originally reported here: https://forums.openrct2.org/topic/4833-bugmiscalculation-with-on-ride-photos/

Quoting:

> I noticed that the calculation for On-Ride Photos makes the assumption that every single guest that rides the ride buys a Photo.
>
> Here it shows that 696 customers ride this ride per hour: https://imgur.com/a/ZtZVa3F
> Here it says that the Income per hour is $1531.20: https://imgur.com/a/F2qwt0u
>
> I had each photo priced at $2.40 which gives $2.20 profit per photo sold. 696 x 2.2 is 1531.20. So the game is calculating that every single guest is buying an On-Ride Photo. In reality 10 people out of 100 have bought an On-Ride Photo so far, so the actual income is more like $153/hour which is lower than the upkeep. The game isn't giving the extra money or anything, it just shows the wrong number for how much money is coming in. There would either need to be a "Customers that bought On-Ride Photo per hour" or after taking the "Customer" value per hour (in this case 696) it would need to be multiplied by the ratio of guests who went on the ride, compared to guests who bought an On-Ride Photo, so in this particular case 100 rode the ride, 10 bought a Photo, so the income off of On-Ride Photo's needs to be divided by 10.

**Comment by tupaschoal** [comment]
Sounds like this could also be the case in the original game, I'd say it needs a bit of investigation

**Comment by duncanspumpkin** [comment]
Yes this is how it is calculated. Its just a rough heuristic and not an actual real value. It basically does (cost of ride + cost of secondary items (photos) - cost to produce secondary items (photos)) * num_customers

**Comment by duncanspumpkin** [comment]
Thinking about this more this is intended behaviour. This is potential profit not last months profit.

**Comment by pizza2004** [comment]
It would at least need to be renamed then, but it seems like it wouldn't be especially useful to have if it can be so drastically different from what's actually going on, since they showed that in this case the ride never turns a profit despite saying it should be making a huge profit.

Perhaps it would be easier to split it into two fields, the actual estimated profit of the last month for admission fees based on the number of customers and the potential profit from secondary items assuming every person bought one, so that no extra math is necessary.  Or a change in logic once the new save format is finished that stores the actual number of secondary items sold so you can do a precise calculation for the total?

**Comment by ocalhoun6** [comment]
![Forest Frontiers 2020-03-16 12-21-37](https://user-images.githubusercontent.com/26573303/76792671-c2ce5b80-6780-11ea-8b71-c84e7864494c.png)

The game already stores On-Ride Photos Sold and Total Customers.

With those two, you can get a ratio of how many customers buy on-ride photos.

Then, multiply that ratio by the number of customers per hour, and you should get a much more accurate (though still not perfect) estimate for the hourly profit from on-ride photos.

**Comment by duncanspumpkin** [comment]
If we are to change it to the real profit per hour i would rather we just saved all the right variables and calculate it correctly.

**Comment by Gymnasiast** [comment]
@duncanspumpkin @tupaschoal I know it is not intended behaviour, because **I** implemented it. There original game didn't even take on-ride photos into account for profit calculation, instead reporting huge losses per hour when the ride was actually turning a profit.

**Comment by pizza2004** [comment]
@ocalhoun6 Yes, but it doesn't store the secondary items per hour like it does the customer, meaning you it is assuming every customer is buying a photo and the best we can estimate is by using the total number of photos vs total customers to get a ratio, but if you increase or decrease the cost of photos that ratio could still be wildly inaccurate for the past hour.  Hence why I suggest we store more data.

**Comment by ocalhoun6** [comment]
Well, yes, storing more data would be ideal, but wouldn't that have to wait for a new save format?
In the meantime, we could use the data already stored to give a better approximation.

**Comment by duncanspumpkin** [comment]
The new save format is very near. Lets say if the new save format is not deployed by 20/04/2020 then we will look into improving the approximation without adding more data.

**Comment by cheweytoo** [comment]
Hm, does this also mean that there's no real way to know whether a ride with on-ride photos is _currently_ profitable or not? Well, except watching "total profit" for a while to see its direction, i guess.

**Comment by Gymnasiast** [comment]
@cheweytoo That's indeed the case.

**Comment by chaitanyathengdi** [comment]
This needs to be worked on, since profit is used when sorting the rides, and this'll give a very different impression when viewed at a glance than what is actually the case.

**Comment by duncanspumpkin** [comment]
@chaitanyathengdi feel free to work on a solution. We plan to address this properly with the new save format.

**Comment by chaitanyathengdi** [comment]
@duncanspumpkin Is there a planned release date for it?

**Comment by chaitanyathengdi** [comment]
I do have an idea about it, let me see how close it gets to the actual.

**Comment by chaitanyathengdi** [comment]
@duncanspumpkin Just basic solution implemented for now, since I didn't find much data in the ride's properties. I'll create more PRs if I find anything to improve.

**Comment by duncanspumpkin** [comment]
I'm keeping this open as a reminder to solve this properly in the new save format.
